pkg/agent/tunnel/tunnel.go

```diff
@@ -75,7 +75,10 @@ func Setup(ctx context.Context, config *config.Node, proxy proxy.Proxy) error {
 
 	endpoint, _ := client.CoreV1().Endpoints("default").Get(ctx, "kubernetes", metav1.GetOptions{})
 	if endpoint != nil {
-		proxy.Update(getAddresses(endpoint))
```

Comment:
Does the `proxy.Update` implementation fail for a zero-length and/or nil slice of addresses? If so, we should fix it there.

My only other guess is you needed to set a breakpoint while debugging.

Reply:
I think it should, the original implementation doesnt have any error check for the update function but this can be fixed